opt_einsum/paths.py

```diff
@@ -761,38 +761,77 @@ def _find_disconnected_subgraphs(inputs, output):
     return subgraphs
 
 
-def _bitmapset_indices(s):
+def _bitmap_select(s, seq):
+    """Select elements of ``seq`` which are marked by the bitmap set ``s``.
+
+    E.g.:
+
+        >>> list(_bitmap_select(0b11010, ['A', 'B', 'C', 'D', 'E']))
+        ['B', 'D', 'E']
     """
-    Returns a generator object allowing to iterate over the elements contained
-    in a bitmap set.
+    return (x for x, b in zip(seq, bin(s)[:1:-1]) if b == '1')
 
-    Parameters
-    ----------
-    s : int
-        The bitmap set to iterate over
 
-    Returns
-    -------
-    path : generator
-        Generator object to iterate over the elements in s
+def _dp_compare_flops(cost1, cost2, i1_union_i2, size_dict, cost_cap, s1, s2, xn, g,
+                      all_tensors, inputs, i1_cut_i2_wo_output, memory_limit, cntrct1, cntrct2):
+    cost = cost1 + cost2 + helpers.compute_size_by_dict(i1_union_i2, size_dict)
+    if cost <= cost_cap:
+        s = s1 | s2
+        if s not in xn or cost < xn[s][1]:
+            # set of remaining tensors (=g-s)
+            r = g & (all_tensors ^ s)
 
-    Examples
-    --------
-    >>> type(_bitmapset_indices(0b1001011))
-    generator
+            # indices of remaining indices:
+            if r:
+                i_r = set.union(*_bitmap_select(r, inputs))
+            else:
+                i_r = set()
+
+            # contraction indices:
+            i_cntrct = i1_cut_i2_wo_output - i_r
```

Comment:
Can we write out `cntrct` (is it contract?), will help with readability.

Reply:
I think it is (I didn't write these variable names originally), but yes can update.

Reply:
Yes, it stands for contract. Don't know why I named it that way.. Maybe some line length issue..